### PR TITLE
Add a script to enable the plugin.

### PR DESCRIPTION
**Pull request by revit13**
Signed-off-by: Revital Sur <eres@il.ibm.com>

### CODE REVIEW

**Comment by roee88** [comment]
Please add a description to the PR. 
It's hard to review PRs without context.

**Comment by revit13** [comment]
This PR adds a script to the Plugin Docker file with commands to enable the plugin and a policy to allow read all the plugin secrets.
This script could be executed during the deployment of the plugin image like in the example: https://github.com/hashicorp/vault-helm/blob/master/values.yaml#L301

**Comment by roee88** [comment]
So at deployment time you can always `vault login root` and do custom installation steps (and then the user needs to disable that user) - ok.

File: startup_plugin.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+vault login root
```
Comment (roee88) [inline startup_plugin.sh:1]: In what conditions the vault binary is available in the image?

Comment (roee88) [inline startup_plugin.sh:1]: Is the user always root or can the user change it in values.yaml?

File: startup_plugin.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+vault secrets enable -path=kubernetes-secrets-reader vault-plugin-secrets-kubernetes-reader
```
Comment (roee88) [inline startup_plugin.sh:2]: Future: if you ever want this plugin to support multicluster then this won't really make sense as you would want to enable it in multiple paths. We don't have such a requirement though.

File: startup_plugin.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+vault policy write read-plugin-secrets - <<EOF
+        path "kubernetes-secrets-reader/*" {
```
Comment (roee88) [inline startup_plugin.sh:4]: Is there a way to enable access only to secrets in a specific namespace (e.g., in M4D when not using builtin catalog you only need the manager to access secrets in m4d-system).

**Review by roee88 — COMMENTED** [review]
~~I don't understand why this needs to be in the docker image and not direct commands you execute within the alpine image.~~

EDIT: unless you think this script is generally useful and not just for your current use in m4d.

**Review by roee88 — APPROVED** [review]

File: startup_plugin.sh (hunk 2 above)
Comment (revit13) [reply startup_plugin.sh:2]: Thats not correct. the auth path should be enabled for each cluster but the backend only once.

**Pull request closed without merging**